tests/Fixtures/Maker/expected/can_create_factory_with_relation_defaults.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace App\Factory;
+
+use Zenstruck\Foundry\ModelFactory;
+use Zenstruck\Foundry\Proxy;
+use Zenstruck\Foundry\Tests\Fixtures\Entity\EntityWithRelations;
+
+/**
+ * @extends ModelFactory<EntityWithRelations>
+ *
+ * @method        EntityWithRelations|Proxy     create(array|callable $attributes = [])
+ * @method static EntityWithRelations|Proxy     createOne(array $attributes = [])
+ * @method static EntityWithRelations|Proxy     find(object|array|mixed $criteria)
+ * @method static EntityWithRelations|Proxy     findOrCreate(array $attributes)
+ * @method static EntityWithRelations|Proxy     first(string $sortedField = 'id')
+ * @method static EntityWithRelations|Proxy     last(string $sortedField = 'id')
+ * @method static EntityWithRelations|Proxy     random(array $attributes = [])
+ * @method static EntityWithRelations|Proxy     randomOrCreate(array $attributes = [])
+ * @method static EntityWithRelations[]|Proxy[] all()
+ * @method static EntityWithRelations[]|Proxy[] createMany(int $number, array|callable $attributes = [])
+ * @method static EntityWithRelations[]|Proxy[] createSequence(array|callable $sequence)
+ * @method static EntityWithRelations[]|Proxy[] findBy(array $attributes)
+ * @method static EntityWithRelations[]|Proxy[] randomRange(int $min, int $max, array $attributes = [])
+ * @method static EntityWithRelations[]|Proxy[] randomSet(int $number, array $attributes = [])
+ */
+final class EntityWithRelationsFactory extends ModelFactory
+{
+    /**
+     * @see https://symfony.com/bundles/ZenstruckFoundryBundle/current/index.html#factories-as-services
+     *
+     * @todo inject services if required
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * @see https://symfony.com/bundles/ZenstruckFoundryBundle/current/index.html#model-factories
+     *
+     * @todo add your default values here
+     */
+    protected function getDefaults(): array
+    {
+        return [
+            'manyToOne' => \Zenstruck\Foundry\Tests\Fixtures\Factories\CategoryFactory::new(),
```

Comment:
Can we have this imported? I think a previous PR of yours played with the imports - can we easily hook into that?

Reply:
yep that's possible, but a little bit tricky: we return a `\Generator<string, string>` from `defaultPropertiesForPersistedObject()`:
```php
// generateFactory()
'defaultProperties' => $this->defaultPropertiesForPersistedObject(),

// defaultPropertiesForPersistedObject()
yield \lcfirst($fieldName) => "\\{$factoryClass}::new(),";
```

I think the cleanest solution would be to introduce some kind of "view model":
```php
class FactoryData
{
    private string $objectShortName;
    private array $uses;
    private array $defaultProperties;
    // ...
}
```

this way we could pass this instance to each functions which would modify it, until we give this object to the template which would use it to display all the needed informations.

WDYT?

Reply:
Yes, that makes sense to me. Want to do that in this PR or another?

Reply:
I think a new PR will be better